runtime/src/lib.rs

```diff
@@ -585,6 +589,30 @@ impl additional_traits::DelegatedDispatchVerifier for Runtime {
 	}
 }
 
+/// Handles gas payment post contract execution (before deferring runtime calls) via CENNZX-Spot exchange.
+impl<T> pallet_contracts::GasHandler<T> for Runtime
+where
+	T: pallet_contracts::Trait + pallet_generic_asset::Trait + crml_cennzx_spot::Trait + Decode,
+{
+	fn empty_unused_gas(transactor: &T::AccountId, gas_meter: GasMeter<T>) {
```

Comment:
How have you made sure that this function would be called for a contract?

Reply:
Thanks for spotting this!